### PR TITLE
Allow Address 2 to be empty

### PR DESCRIPTION
**Pull request by Abdalsalaam**
https://wordpress.org/support/topic/shipping-address-2-is-empty-error-message-2

### CODE REVIEW

File: pr-dhl-woocommerce/includes/pr-dhl-api/class-pr-dhl-api-soap-label.php [hunk 1]
```diff
@@ -463,38 +463,40 @@ protected function set_arguments( $args ) {
...
+					if( $set_key === false ) {
 						$set_key = $address_key;
 					}
-				}
 
-				// If no number was found, then take last part of address no matter what it is
-				if( $set_key === false ) {
-					$set_key = $address_key;
+					// The number is the first part of address 1
+					if( $set_key == 0 ) {
```
Comment (iyut) [inline pr-dhl-woocommerce/includes/pr-dhl-api/class-pr-dhl-api-soap-label.php:488]:
```suggestion
					if ( 0 === $set_key ) {
```

File: pr-dhl-woocommerce/includes/pr-dhl-api/class-pr-dhl-api-soap-label.php [hunk 2]
```diff
@@ -463,38 +463,40 @@ protected function set_arguments( $args ) {
...
+					// ...last found number will be 'address_2'
+					foreach ($address_exploded as $address_key => $address_value) {
+						if (is_numeric($address_value)) {
+							// Set last index as street number
+							$set_key = $address_key;
+						}
+					}
+
+					// If no number was found, then take last part of address no matter what it is
+					if( $set_key === false ) {
```
Comment (iyut) [inline pr-dhl-woocommerce/includes/pr-dhl-api/class-pr-dhl-api-soap-label.php:483]:
```suggestion
					if ( false === $set_key ) {
```

File: pr-dhl-woocommerce/includes/pr-dhl-api/class-pr-dhl-api-soap-label.php [hunk 3]
```diff
@@ -463,38 +463,40 @@ protected function set_arguments( $args ) {
...
-				// Loop through address and set number value only...
-				// ...last found number will be 'address_2'
-				foreach ($address_exploded as $address_key => $address_value) {
-					if (is_numeric($address_value)) {
-						// Set last index as street number
+				if( count($address_exploded) > 1 ) {
+					// Loop through address and set number value only...
+					// ...last found number will be 'address_2'
+					foreach ($address_exploded as $address_key => $address_value) {
+						if (is_numeric($address_value)) {
```
Comment (iyut) [inline pr-dhl-woocommerce/includes/pr-dhl-api/class-pr-dhl-api-soap-label.php:476]:
```suggestion
						if ( is_numeric( $address_value ) ) {
```

File: pr-dhl-woocommerce/includes/pr-dhl-api/class-pr-dhl-api-soap-label.php [hunk 4]
```diff
@@ -463,38 +463,40 @@ protected function set_arguments( $args ) {
...
 
-				// Loop through address and set number value only...
-				// ...last found number will be 'address_2'
-				foreach ($address_exploded as $address_key => $address_value) {
-					if (is_numeric($address_value)) {
-						// Set last index as street number
+				if( count($address_exploded) > 1 ) {
+					// Loop through address and set number value only...
+					// ...last found number will be 'address_2'
+					foreach ($address_exploded as $address_key => $address_value) {
```
Comment (iyut) [inline pr-dhl-woocommerce/includes/pr-dhl-api/class-pr-dhl-api-soap-label.php:475]:
```suggestion
					foreach ( $address_exploded as $address_key => $address_value ) {
```

File: pr-dhl-woocommerce/includes/pr-dhl-api/class-pr-dhl-api-soap-label.php [hunk 5]
```diff
@@ -463,38 +463,40 @@ protected function set_arguments( $args ) {
                 if( count($address_exploded) == 1 ) {
                     // Break address into pieces by '.'
                     $address_exploded = explode('.', $args['shipping_address']['address_1']);
-
-                    if( count($address_exploded) == 1 ) {
+					// Check if address 2 empty and its in Germany
+                    if( count($address_exploded) == 1 && 'DE' === $args['shipping_address']['country'] ) {
```
Comment (iyut) [inline pr-dhl-woocommerce/includes/pr-dhl-api/class-pr-dhl-api-soap-label.php:467]:
```suggestion
                    if ( 1=== count( $address_exploded ) && 'DE' === $args['shipping_address']['country'] ) {
```

File: pr-dhl-woocommerce/includes/pr-dhl-api/class-pr-dhl-api-soap-label.php [hunk 6]
```diff
@@ -463,38 +463,40 @@ protected function set_arguments( $args ) {
...
                         throw new Exception(__('Shipping "Address 2" is empty!', 'dhl-for-woocommerce'));
                     }
                 }
 
-				// Loop through address and set number value only...
-				// ...last found number will be 'address_2'
-				foreach ($address_exploded as $address_key => $address_value) {
-					if (is_numeric($address_value)) {
-						// Set last index as street number
+				if( count($address_exploded) > 1 ) {
```
Comment (iyut) [inline pr-dhl-woocommerce/includes/pr-dhl-api/class-pr-dhl-api-soap-label.php:472]:
```suggestion
				if ( count( $address_exploded ) > 1 ) {
```

**Review by iyut — CHANGES_REQUESTED** [review]
So sorry for being picky here. Basically, just need to change some spacing.

**Comment by Abdalsalaam** [comment]
> So sorry for being picky here. Basically, just need to change some spacing.

thank you! I prefer these review

**Review by iyut — APPROVED** [review]
LGTM!

**Pull request closed without merging**